text/0000-transaction-status.md

```diff
@@ -0,0 +1,118 @@
+This is displayed as a list where items are gradually checked out once they are completed, e.g.:
+* Transaction 04ffa4f received by a node;
+* Transaction 04ffa4f:
+    * Received by a chunk producer on shard 93667aa;
+    * Included in chunk (1 receipt created: 20693fc);
+    * Included in block;
+* Receipt 20693fc:
+    * Received by a chunk producer on shard 9227d15;
+    * Included in chunk (1 refund created: 1ce44a3);
+    * Included in block;
+* Refund 1ce44a3:
+    * Received by a chunk producer on shard 93667aa;
+    * Included in chunk;
+    * Included in block;
+* Complete.
+
+The UI has the following properties: 
+* a transaction can be first received by a non-validating node that reroutes it to the validator;
+* a transaction can result in multiple receipts, callbacks, refunds, and other transactions;
+* these other transactions can be executed on different shards and asynchronously, which means the items might be checked out out of order;
+* the node the user uses to query for the status might not have a perfect synchronization with the network and so multiple items will be checked out at once, instead of one at a time.
+* the ids of the transactions and shards a truncated to the first 7 base64 characters, similarly to how github does it.
+
+Similar UI is available from the command line tools, e.g. near shell interactively prints the very same list, similar to how advanced progress bar in shell tools work.
+Near shell supports several levels of verbosity, where the highest level includes full base64 identifiers of everything: the node that received a transaction, hashes of the chunk and the block.
+
+Finally, the RPC of a node is the source of this information and it allows querying it in a structured format.
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+The source of transaction status information is the RPC of a node. Both validating and non-validating nodes should have
+access to this information. However, non-validating nodes might have a slightly stale information and they might discover
+about transaction included in chunk simultaneously with it included in the block. So the status known to non-validating
+might jump several items at once, e.g. from "received" to "included in block", or even to "completed".
+
+The structured information returned by an RPC should be exhaustive and forward compatible to avoid breaking wallet and Near shell
+in the future because of the changes in the structure of this format. Wallet and Near shell, however, should also be forward compatible,
+meaning they should not fail if some field is missing from the structured output of RPC.
+
+To simplify the development RPC `tx_details` should return the maximum exhaustive information. And we do not
+create overloads of this RPC based on verbosity. So the following RPC call `tx_details` with `{"tx_hash": "04ffa4f"}`
+will return the following (notice how we allow incomplete hash in the RPC):
+```json
+[
```

Comment:
This format doesn't really make sense to me.
Who are you going to query to get this information? Are we going to query the whole network to find which nodes see what?
In reality transaction is in:
  - Unknown state
  - In mempool of the given node we query. Here we may have extended mempool for non-validating nodes (right now we don't have that), just so that this nodes can resend transactions if they were not received by current validators (right now it's up to the end client to resend). But the node doesn't really know if it was received (anything that validators says it's not enforceable anyway), so they must decide to resend them based on received chunks back. There is also caveat that this node may not track the shard this TX is aimed at, meaning it actually may not know at all if this Tx was included.
  - included in chunk
  - included in block
And then for each receipt DFS:
  - receipt included in chunk (e.g. we transition into this, when receipts from previously produced chunks actually got processed by another chard)
  - receipt included in block